core/tools/color.js

```diff
@@ -358,20 +417,21 @@
 					alpha = Number( alpha.toFixed( 1 ) );
 				}
 
-				return [
-					hexToNumber( parts[ 1 ] + parts[ 2 ] ),
-					hexToNumber( parts[ 3 ] + parts[ 4 ] ),
-					hexToNumber( parts[ 5 ] + parts [ 6 ] ),
-					alpha
-				];
+				return {
+					type: CKEDITOR.tools.color.TYPE_RGB,
```

Comment:
It looks a bit awkward that we return `TYPE_RGB` instead of `TYPE_HEX` here. I know that functionally it won't make any difference, but maybe we could distinguish them for clarity?

Reply:
Unless it actually converts the hex color to RGB (and I guess it does) 🤔

Reply:
Technically all colors are converted to RGB. However hex is only the other representation of RGB color, while HSL is a completely different color format. Because of that I've introduced only two types – RGB (for both representations, `rgb()` and hex ones) and HSL.

Reply:
Ok, makes sense.